### PR TITLE
HereNow changed

### PR DESCRIPTION
**Pull request by Chesteer89**
Occupancy Service changed,
Channel Service changed,

### CODE REVIEW

**Review by mateusz-cichosz — APPROVED** [review]

File: chat-components/src/main/java/com/pubnub/components/chat/service/channel/OccupancyService.kt [hunk 1]
```diff
@@ -2,13 +2,14 @@ package com.pubnub.components.chat.service.channel
...
 import com.pubnub.components.chat.ui.component.presence.Presence
+import com.pubnub.framework.data.ChannelGroupId
 import com.pubnub.framework.data.ChannelId
 import com.pubnub.framework.data.Occupancy
 import com.pubnub.framework.data.UserId
 import kotlinx.coroutines.flow.Flow
 
 interface OccupancyService {
-    fun bind()
+    fun bind(channels: List<ChannelId> = emptyList(), channelGroups: List<ChannelGroupId> = emptyList())
```
Comment (Chesteer89) [inline chat-components/src/main/java/com/pubnub/components/chat/service/channel/OccupancyService.kt:12]: @kazydek `OccupancyService.bind` method takes two lists now

File: chat-components/src/main/java/com/pubnub/components/chat/service/channel/DefaultChannelService.kt [hunk 2]
```diff
@@ -2,23 +2,41 @@ package com.pubnub.components.chat.service.channel
...
-        subscribedChannels = listOf(*channels)
-        pubNub.subscribe(channels = subscribedChannels, withPresence = true)
+    /**
+     * Subscribes to channels and channel groups
+     *
+     * @param channels - Channels to subscribe/unsubscribe. Either channel or channelGroups are required.
+     * @param channelGroups - Channel groups to subscribe/unsubscribe. Either channelGroups or channels are required.
+     * @param withPresence - Also subscribe to related presence channel.
+     */
+    override fun bind(channels: List<ChannelId>, channelGroups: List<ChannelGroupId>, withPresence: Boolean) {
```
Comment (Chesteer89) [inline chat-components/src/main/java/com/pubnub/components/chat/service/channel/DefaultChannelService.kt:24]: @kazydek Default service updated

File: chat-components/src/main/java/com/pubnub/components/chat/provider/ChatProvider.kt [hunk 3]
```diff
@@ -314,14 +312,15 @@ fun Synchronize() {
     val channels by membershipRepository.getAll(currentUser).collectAsState(initial = emptyList())
     val errorHandler = LocalLogger.current
     DisposableEffect(channels) {
-        val channelArray = channels.map { it.channelId }.toTypedArray()
-        errorHandler.d("Bind for channels ${channelArray.joinToString()}")
+        val channelList = channels.map { it.channelId }
```
[the remark this answers is not in the document]
Comment (kazydek) [reply chat-components/src/main/java/com/pubnub/components/chat/provider/ChatProvider.kt:315]: I checked, and that info isn't in the docs, so no update is required.

File: chat-components/src/main/java/com/pubnub/components/chat/service/channel/DefaultChannelService.kt (hunk 2 above)
Comment (kazydek) [reply chat-components/src/main/java/com/pubnub/components/chat/service/channel/DefaultChannelService.kt:24]: No update is needed (we don't have that info in the docs)

File: chat-components/src/main/java/com/pubnub/components/chat/service/channel/ChannelService.kt [hunk 4]
```diff
@@ -3,9 +3,11 @@ package com.pubnub.components.chat.service.channel
 import androidx.compose.runtime.compositionLocalOf
 import com.pubnub.components.data.channel.Channel
 import com.pubnub.components.data.channel.DBChannel
+import com.pubnub.framework.data.ChannelGroupId
+import com.pubnub.framework.data.ChannelId
 
 interface ChannelService<Data : Channel> {
-    fun bind(vararg channels: String)
+    fun bind(channels: List<ChannelId> = emptyList(), channelGroups: List<ChannelGroupId> = emptyList(), withPresence: Boolean = false)
```
[the remark this answers is not in the document]
Comment (kazydek) [reply chat-components/src/main/java/com/pubnub/components/chat/service/channel/ChannelService.kt:10]: Done

File: chat-components/src/main/java/com/pubnub/components/chat/service/channel/OccupancyService.kt (hunk 1 above)
Comment (kazydek) [reply chat-components/src/main/java/com/pubnub/components/chat/service/channel/OccupancyService.kt:12]: There's no such detailed info on it in the docs.

**Pull request merged**